horace_core/sqw/PixelData/@pix_metadata/pix_metadata.m

```diff
@@ -58,6 +58,20 @@
             end
 
         end
+        function is = is_range_valid(obj,fld)
+            % check if the range for the appropriate fields, provided as
+            % input is valid, i.e. not equal to empty range;
+            if nargin == 1 % check the whole range
+                idx = PixelDataBase.FIELD_INDEX_MAP('all');
+            elseif iscell(fld)
+                idx = cellfun(@(fl)PixelDataBase.FIELD_INDEX_MAP(fl),fld);
+            else
+                idx = PixelDataBase.FIELD_INDEX_MAP(fld);
+            end
+            invalid = obj.data_range_(:,idx) == PixelDataBase.EMPTY_RANGE(:,idx);
+            is = ~any(invalid(:));
```

Comment:
Minor thing and subjective, but given the function is in the positive would probably find it easier to rationalise if this were in the positive too:
```suggestion
            valid = obj.data_range_(:,idx) ~= PixelDataBase.EMPTY_RANGE(:,idx);
            is = all(valid(:));
```